File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ContactService.cs
﻿using System.Collections.Generic;
using System.Linq;
using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
using GovUk.Education.ExploreEducationStatistics.Content.Model;
using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;

namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
{
    public class ContactService : IContactService
    {
        private readonly ContentDbContext _context;

        public ContactService(ContentDbContext context)
        {
            _context = context;
        }

        public List<Contact> GetContacts()
        {
            return _context.Contacts.OrderBy(c => c.TeamName).ThenBy(c => c.ContactName).ToList();
        }
    }
}

File: src/explore-education-statistics-admin/src/services/contactService.ts
import client from '@admin/services/utils/service';

export interface ContactDetails {
  id: string;
  contactName: string;
  contactTelNo: string;
  teamEmail: string;
  teamName?: string;
}

const contactService = {
  getContacts(): Promise<ContactDetails[]> {
    return client.get<ContactDetails[]>('/contacts');
  },
};

export default contactService;
